clientgui/DlgAdvPreferencesBase.cpp

```diff
@@ -250,233 +249,296 @@ wxPanel* CDlgAdvPreferencesBase::createProcessorTab(wxNotebook* notebook)
     );
     m_chkGPUProcInUse->SetToolTip(
         _("Check this to suspend GPU computing when you're using the computer.")
     );
-    suspendComputingBoxSizer->Add( m_chkGPUProcInUse, 0, wxALL, 5 );
-
-    // min idle time
-    wxString ProcIdleForTT(_("This determines when the computer is considered 'in use'."));
+    box_sizer->Add( m_chkGPUProcInUse, 0, wxALL, 5 );
 
-    wxStaticText* staticText24 = new wxStaticText(
-            suspendComputingStaticBox, ID_DEFAULT,
-            // context: 'In use' means mouse/keyboard input in last ___ minutes
-            _("'In use' means mouse/keyboard input in last"),
-            wxDefaultPosition, wxDefaultSize, 0
-        );
+    // max # CPUs
+    //
+    /*xgettext:no-c-format*/
+    wxString MaxCPUPctTT(_("Keep some CPUs free for other applications. Example: 75% means use 6 cores on an 8-core CPU."));
+    wxStaticText* staticText20 = new wxStaticText(
+        box, ID_DEFAULT, _("Use at most"), wxDefaultPosition, wxDefaultSize, 0
+    );
+    m_txtProcUseProcessors = new wxTextCtrl(box, ID_TXTPROCUSEPROCESSORS, wxEmptyString, wxDefaultPosition, textCtrlSize, wxTE_RIGHT );
+    /*xgettext:no-c-format*/
+    wxStaticText* staticText21 = new wxStaticText(box, ID_DEFAULT, _("% of the CPUs"), wxDefaultPosition, wxDefaultSize, 0 );
+    addNewRowToSizer(box_sizer, MaxCPUPctTT, staticText20, m_txtProcUseProcessors, staticText21);
 
-    m_txtProcIdleFor = new wxTextCtrl(
-        suspendComputingStaticBox, ID_TXTPROCIDLEFOR, wxEmptyString, wxDefaultPosition, getTextCtrlSize(wxT("999.99")), wxTE_RIGHT
+    // CPU throttling
+    //
+    /*xgettext:no-c-format*/
+    wxString MaxCPUTimeTT(_("Suspend/resume computing every few seconds to reduce CPU temperature and energy usage. Example: 75% means compute for 3 seconds, wait for 1 second, and repeat."));
+    wxStaticText* staticText22 = new wxStaticText(
+        box, ID_DEFAULT, _("Use at most"), wxDefaultPosition, wxDefaultSize, 0
     );
+    m_txtProcUseCPUTime = new wxTextCtrl(box, ID_TXTPROCUSECPUTIME, wxEmptyString, wxDefaultPosition, textCtrlSize, wxTE_RIGHT );
+    /*xgettext:no-c-format*/
+    wxStaticText* staticText23 = new wxStaticText(box, ID_DEFAULT, _("% of CPU time"), wxDefaultPosition, wxDefaultSize, 0 );
+    addNewRowToSizer(box_sizer, MaxCPUTimeTT, staticText22, m_txtProcUseCPUTime, staticText23);
 
-    wxStaticText* staticText25 = new wxStaticText(
-        suspendComputingStaticBox, ID_DEFAULT,
-        // context: 'In use' means mouse/keyboard input in last ___ minutes
-        _("minutes"),
+    // max CPU load
+    //
+    m_chkMaxLoad = new wxCheckBox(
+        box, ID_CHKMAXLOAD,
+        wxString(_("Suspend when non-BOINC CPU usage is above")),
         wxDefaultPosition, wxDefaultSize, 0
     );
+    m_txtMaxLoad = new wxTextCtrl(
+        box, ID_TXTMAXLOAD, wxEmptyString, wxDefaultPosition, getTextCtrlSize(wxT("100.00")), wxTE_RIGHT
+    );
+    addNewRowToSizer(box_sizer,
```

Comment:
To eliminate the compiler syntax error, you must create a separate variable with this string thus:
`    wxString SuspendOnBusyTT(_("Suspend computing when your computer is busy running other programs."));`
and pass `SuspendOnBusyTT ` in the call.